Hestia/app/src/main/java/hestia/backend/Activator.java

```diff
@@ -3,94 +3,83 @@
 /**
  * This class represents a single activator on a device. A single device can have multiple activators.
  * The activator has an id so we can reference it on the server.
- * Furthermore there is a string name and a string state type,
- * which is currently used for casting the generic state variable to an actual type.
+ * Furthermore there is a string name and a string state name,
+ * which is currently used for casting the generic state variable to an actual name.
  * @see Device
  * @see ActivatorState
  */
 public class Activator {
-    private int id;
+    private String activatorId;
+    private int rank;
     private ActivatorState state;
     private String name;
 
     /**
      * Creates an Activator with the specified id, state and name.
```

Comment:
update your documentation with the code.
Here you are not explaining what 'rank' is for. Automatic coverage reports should help you see these mistakes better.